File: packages/python/src/endpoints/rules.py
"""
Rules API Client
"""

from typing import Dict, List, Optional, Any
import httpx


class RulesClient:
    """Client for rules endpoints."""

    def __init__(self, base_url: str, client: Optional[httpx.AsyncClient] = None):
        self.base_url = base_url.rstrip('/')
        self.client = client

    async def _request(self, method: str, path: str, **kwargs) -> Any:
        """Make HTTP request."""
        url = self.base_url + path
        if self.client:
            response = await self.client.request(method, url, **kwargs)
            response.raise_for_status()
            return response.json()
        else:
            async with httpx.AsyncClient() as client:
                response = await client.request(method, url, **kwargs)
                response.raise_for_status()
                return response.json()


    async def get_bundles(self, category: Optional[str] = None, official_only: Optional[bool] = None, page: Optional[float] = None, per_page: Optional[float] = None):
        """GET /bundles"""
        url = f"/bundles"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def post_bundles_bundle_id_install(self, bundle_id: str, installation: Any):
        """POST /bundles/{bundle_id}/install"""
        url = f"/bundles/{bundle_id}/install"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("POST", url, params=params, json=data)


    async def get_bundles_bundle_id_rules(self, bundle_id: str):
        """GET /bundles/{bundle_id}/rules"""
        url = f"/bundles/{bundle_id}/rules"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def post_validate(self):
        """POST /validate"""
        url = f"/validate"
        params = {}
        data = None
        return await self._request("POST", url, params=params, json=data)


    async def get_updates(self):
        """GET /updates"""
        url = f"/updates"
        params = {}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def delete_bundles_bundle_id_install(self, bundle_id: str):
        """DELETE /bundles/{bundle_id}/install"""
        url = f"/bundles/{bundle_id}/install"
        params = {}
        data = bundle_id
        return await self._request("DELETE", url, params=params, json=data)


    async def get_organizations_organization_id_bundles(self, organization_id: str):
        """GET /organizations/{organization_id}/bundles"""
        url = f"/organizations/{organization_id}/bundles"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def post_organizations_organization_id_bundles_bundle_id_install(self, organization_id: str, bundle_id: str):
        """POST /organizations/{organization_id}/bundles/{bundle_id}/install"""
        url = f"/organizations/{organization_id}/bundles/{bundle_id}/install"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("POST", url, params=params, json=data)


    async def delete_organizations_organization_id_bundles_bundle_id(self, organization_id: str, bundle_id: str):
        """DELETE /organizations/{organization_id}/bundles/{bundle_id}"""
        url = f"/organizations/{organization_id}/bundles/{bundle_id}"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("DELETE", url, params=params, json=data)


    async def get_organizations_organization_id_rules(self, organization_id: str, category: Optional[str] = None, severity: Optional[str] = None):
        """GET /organizations/{organization_id}/rules"""
        url = f"/organizations/{organization_id}/rules"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def get_organizations_organization_id_rules_stats(self, organization_id: str):
        """GET /organizations/{organization_id}/rules/stats"""
        url = f"/organizations/{organization_id}/rules/stats"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)

File: packages/typescript/src/endpoints/rules.ts
/**
 * Rules API Client
 */

import { AxiosInstance } from 'axios'

export class RulesClient {
  private httpClient: AxiosInstance

  constructor(private baseURL: string, httpClient: AxiosInstance) {
    this.httpClient = httpClient
  }

  /**
   * GET /bundles
   */
  async getBundles(category?: string, official_only?: boolean, page?: number, per_page?: number): Promise<RuleBundleList> {
        const params = {'category: category, official_only: official_only, page: page, per_page: per_page'}
        const data = undefined
        return this.httpClient.get(`/bundles`, params)
  }
  /**
   * POST /bundles/{bundle_id}/install
   */
  async postBundles{bundle_id}install(bundle_id: string, installation: RuleBundleInstallationCreate): Promise<RuleBundleInstallationSchema> {
        const params = {}
        const data = {'bundle_id: bundle_id, installation: installation'}
        return this.httpClient.post(`/bundles/{bundle_id}/install`, data, params)
  }
  /**
   * GET /bundles/{bundle_id}/rules
   */
  async getBundles{bundle_id}rules(bundle_id: string): Promise<List[RuleSchema]> {
        const params = {'bundle_id: bundle_id'}
        const data = undefined
        return this.httpClient.get(`/bundles/{bundle_id}/rules`, params)
  }
  /**
   * POST /validate
   */
  async postValidate(): Promise<RuleValidationResult> {
        const params = {}
        const data = undefined
        return this.httpClient.post(`/validate`, data, params)
  }
  /**
   * GET /updates
   */
  async getUpdates(): Promise<List[RuleBundleUpdates]> {
        const params = {}
        const data = undefined
        return this.httpClient.get(`/updates`, params)
  }
  /**
   * DELETE /bundles/{bundle_id}/install
   */
  async deleteBundles{bundle_id}install(bundle_id: string): Promise<any> {
        const params = {}
        const data = bundle_id
        return this.httpClient.delete(`/bundles/{bundle_id}/install`, data, params)
  }
  /**
   * GET /organizations/{organization_id}/bundles
   */
  async getOrganizations{organization_id}bundles(organization_id: string): Promise<List[RuleBundleSchema]> {
        const params = {'organization_id: organization_id'}
        const data = undefined
        return this.httpClient.get(`/organizations/{organization_id}/bundles`, params)
  }
  /**
   * POST /organizations/{organization_id}/bundles/{bundle_id}/install
   */
  async postOrganizations{organization_id}bundles{bundle_id}install(organization_id: string, bundle_id: string): Promise<any> {
        const params = {}
        const data = {'organization_id: organization_id, bundle_id: bundle_id'}
        return this.httpClient.post(`/organizations/{organization_id}/bundles/{bundle_id}/install`, data, params)
  }
  /**
   * DELETE /organizations/{organization_id}/bundles/{bundle_id}
   */
  async deleteOrganizations{organization_id}bundles{bundle_id}(organization_id: string, bundle_id: string): Promise<any> {
        const params = {}
        const data = {'organization_id: organization_id, bundle_id: bundle_id'}
        return this.httpClient.delete(`/organizations/{organization_id}/bundles/{bundle_id}`, data, params)
  }
  /**
   * GET /organizations/{organization_id}/rules
   */
  async getOrganizations{organization_id}rules(organization_id: string, category?: string, severity?: string): Promise<List[RuleSchema]> {
        const params = {'organization_id: organization_id, category: category, severity: severity'}
        const data = undefined
        return this.httpClient.get(`/organizations/{organization_id}/rules`, params)
  }
  /**
   * GET /organizations/{organization_id}/rules/stats
   */
  async getOrganizations{organization_id}rulesstats(organization_id: string): Promise<any> {
        const params = {'organization_id: organization_id'}
        const data = undefined
        return this.httpClient.get(`/organizations/{organization_id}/rules/stats`, params)
  }
}
